### PR TITLE
Updated to stable versions of Httplug

### PR DESCRIPTION
**Pull request by Nyholm**
This PR is a cleanup and using stable versions of Httplug. 

I would prefer to make a first tag after this PR so we can use it for tests in HttplugBundle.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -12,14 +12,17 @@
...
-        "php-http/client-common": "^0.1.1",
+        "php-http/httplug": "^1.0",
         "react/http-client": "^0.4.8",
         "react/dns": "^0.4.1",
         "react/stream": "^0.4.3"
     },
     "require-dev": {
         "guzzlehttp/psr7": "^1.0",
-        "php-http/adapter-integration-tests": "dev-master"
+        "php-http/client-common": "^1.0",
```
Comment (sagikazarmark) [inline composer.json]: Why is this a dev dependency?

File: composer.json [hunk 2]
```diff
@@ -12,14 +12,17 @@
...
+        "php-http/message-factory": "^1.0",
```
Comment (sagikazarmark) [inline composer.json]: Same question here

File: .travis.yml [hunk 3]
```diff
@@ -31,6 +31,7 @@ matrix:
 
 before_install:
     - travis_retry composer self-update
+    - wget https://github.com/puli/cli/releases/download/1.0.0-beta10/puli.phar && chmod +x puli.phar
```
Comment (sagikazarmark) [inline .travis.yml]: I think discovery should be added to the list of dependencies. Then we wouldn't need this.

File: composer.json (hunk 1 above)
Comment (Nyholm) [inline composer.json]: Sorry, that is wrong

File: .travis.yml (hunk 3 above)
Comment (Nyholm) [inline .travis.yml]: Discovery is only used by the integration tests and it is an indirect dependency. It failed locally if I do not have puli.phar installed.

File: composer.json (hunk 2 above)
Comment (Nyholm) [inline composer.json]: We use it for the tests: https://github.com/php-http/react-adapter/pull/10/files#diff-4bad3a20665cbb8360f524e159537fe9R8

Comment (Nyholm) [inline composer.json]: But it will be required by integration-tests -> discovery -> message-factory. 
I can remove it.

Comment (Nyholm) [inline composer.json]: Or use discovery here as well..

Comment (sagikazarmark) [inline composer.json]: If the message factory is used here, then it should be a requirement.

File: .travis.yml (hunk 3 above)
Comment (sagikazarmark) [inline .travis.yml]: That's strange. Maybe it is because the binaries are not installed properly? Can you try removing vendor locally and reinstalling dependencies? Also, I wonder why discovery is not used in this client.

To make it work, MessageFactory should become an optional dependency.

Comment (sagikazarmark) [inline .travis.yml]: Not sure if it makes sense though, since there are other, non-optional dependencies, so this client cannot be discovered anyway.

Comment (Nyholm) [inline .travis.yml]: I believe I was mistaken. 
I tried your suggestion to remove the puli.phar and it works as you expected

**Comment by Nyholm** [comment]
Thank you for reviewing this so quickly. I have addressed your comments and Im waiting for Travis to run the tests.

**Comment by Nyholm** [comment]
There. The tests passes

File: composer.json [hunk 4]
```diff
@@ -12,14 +12,15 @@
     ],
     "require": {
         "php": ">=5.4",
-        "php-http/client-common": "^0.1.1",
```
Comment (sagikazarmark) [inline composer.json]: I meant, why is this removed? Is this not necessary?

Comment (Nyholm) [inline composer.json]: No, it is never used.

**Comment by sagikazarmark** [comment]
Hm, it seems the react client might be discoverable after all. Let's merge this, I will add a PR for discovery stuff.
